### PR TITLE
Update FAQ

### PR DESCRIPTION
**Pull request by danintel**
* Add B9Lab online Sawtooth developer course
* Add recommended PoET settings for small test networks
* Mention /etc/sawtooth/path.toml.example for directory configuration
* Add Consensus Engine SDK slides for the 20180426 Tech Forum video
* Add answers for consensus engine starting order
* Add back pressure test answer
* Add ``failed state root hash validation`` answer
* Add corrections for App Dev Guide Ubuntu install instructions
* Add PoET fork answer
* Add key file not readable answer
* Add file permission answer
* Add installation answer for nightly on Bionic
* Validator default is now parallel mode
* Add PoET create wait certificate error answer
* Add PoET BFT answer
* Add Grafana video
* Add PoET CFT production use qualification
* Add Rust SDK development for MS Windows
* Add Sawtooth 1.2 Ubuntu Devmode installation instructions
* Add consensus engine settings
* Add nightly package install workaround for Bionic
* Add Certified Hyperledger Sawtooth Administrator (CHSA) information
* Add Sawtooth Explorer instructions
* Update PBFT settings

Signed-off-by: danintel <daniel.anderson@intel.com>

### CODE REVIEW

File: generator/source/faq/consensus.rst
Comment (TomBarnes) [inline generator/source/faq/consensus.rst]: I would say that PoET CFT is able to run on systems that do not have SGX and is suitable for use in systems that do not require BFT,

**Review by TomBarnes — APPROVED** [review]
Other than one comment regarding PoET /SGX requirement, looks good.

File: generator/source/faq/consensus.rst
Comment (danintel) [reply generator/source/faq/consensus.rst]: OK. I added the qualifier "for systems that do not require BFT" to the first sentence.

File: generator/source/faq/client.rst
Comment (vaporos) [inline generator/source/faq/client.rst]: I don't think we need the third-party qualifier here.

File: generator/source/faq/consensus.rst
Comment (vaporos) [inline generator/source/faq/consensus.rst]: Remove the Fabric comparison - the mechanisms are very different, and that makes this comparison potentially confusing.

Remove the reference about production networks. If mentioning BFT, maybe say something like "Note that PoET CFT is not Byzantine fault tolerant."

File: generator/source/faq/consensus.rst
Comment (vaporos) [inline generator/source/faq/consensus.rst:26]: Supported in Sawtooth 1.0 only - should that be mentioned here?

File: generator/source/faq/consensus.rst
Comment (vaporos) [inline generator/source/faq/consensus.rst:22]: Can we omit the versions so this stays up-to-date easier?)

File: generator/source/faq/consensus.rst
Comment (vaporos) [inline generator/source/faq/consensus.rst]: I think we should take a position on its production-readiness from a maturity standpoint but make it clear separately that its a business and trust decision whether CFT is good enough for production. (My opinion is generally no -- if CFT is good enough (which means a lack of adversarial trust issues being solved), then why use blockchain?)

File: generator/source/faq/consensus.rst
Comment (vaporos) [inline generator/source/faq/consensus.rst]: Line break issues here - the e should be on the previous line, and the s from the previous line should be on this line.

File: generator/source/faq/consensus.rst
Comment (vaporos) [inline generator/source/faq/consensus.rst]: 5 might be a better recommendation. Even with 10, it can be hard to keep from tripping Z test. But 5 gives better wait time averages.

File: generator/source/faq/consensus.rst
Comment (vaporos) [inline generator/source/faq/consensus.rst:223]: This could be strengthened to "We recommend PBFT for small networks." Raft is less interesting being CFT and not BFT, and having overall less testing done.

File: generator/source/faq/consensus.rst
Comment (vaporos) [inline generator/source/faq/consensus.rst]: This should be considered a bug. Maybe we could link to a JIRA here.

File: generator/source/faq/installation.rst
Comment (vaporos) [inline generator/source/faq/installation.rst]: Is this a typo "config-genesispoet_genesis.batch"?

File: generator/source/faq/installation.rst
Comment (vaporos) [inline generator/source/faq/installation.rst]: We should not steer people to the old 0.8 version at all, but instead focus on installing it with 1.1.

File: generator/source/faq/installation.rst
Comment (vaporos) [inline generator/source/faq/installation.rst]: We probably consider the necessity of this patch a bug?

File: generator/source/faq/installation.rst
Comment (vaporos) [inline generator/source/faq/installation.rst:265]: This should also be tied to some bug, otherwise we won't know when to remove the workaround from the FAQ.

File: generator/source/faq/rest.rst
Comment (vaporos) [inline generator/source/faq/rest.rst]: This might be partially incorrect in that the back pressure isn't just about slower members of the network. Rather, it is about rejecting load (in the form of batches) that the network wouldn't be able to accommodate in a reasonable amount of time (next couple blocks).

This should never be turned off as without this feature it is trivial to do a client-based attack which overwhelms the network creating a DoS.

File: generator/source/faq/settings.rst
Comment (vaporos) [inline generator/source/faq/settings.rst]: For best practice - we should avoid referring to '1.2' or other unreleased versions - we may never release a 1.2, we could go instead to 2.0. This sounds like a compatibility bug to me.

File: generator/source/faq/transaction-processing.rst
Comment (vaporos) [inline generator/source/faq/transaction-processing.rst]: Typo 'generateing'

File: generator/source/faq/validator.rst
Comment (vaporos) [inline generator/source/faq/validator.rst]: Same comment here re:1.2 - "and earlier and ``parallel`` for nightly master" or something.

File: generator/source/faq/transaction-processing.rst
Comment (danintel) [reply generator/source/faq/transaction-processing.rst]: Accepted.

File: generator/source/faq/client.rst
Comment (danintel) [reply generator/source/faq/client.rst]: Accepted.

File: generator/source/faq/consensus.rst
Comment (danintel) [reply generator/source/faq/consensus.rst]: Accepted.
I used the phrase for PoET CFT "Has Crash Fault Tolerance (CFT), but is not Byzantine Fault Tolerant (BFT)"

File: generator/source/faq/consensus.rst
Comment (danintel) [reply generator/source/faq/consensus.rst:26]: Accepted. I'll add "Currently supported in Sawtooth 1.0 only."

File: generator/source/faq/consensus.rst
Comment (danintel) [reply generator/source/faq/consensus.rst:22]: I want to leave the consensus versions for now as they are mandatory for Sawtooth 1.2 and are difficult-to-find in documentation.

File: generator/source/faq/consensus.rst
Comment (danintel) [reply generator/source/faq/consensus.rst]: Accepted. I'll remove mention of production-readiness. Here is the reworded answer:

> What protections does PoET CFT have, since it is not BFT?
> ----------------------------------------
> It is for systems that do not have SGX and do not require BFT. Both PoET CFT and PoET SGX 
have tests to guard against bad actors, such as the "Z Test" to check a validator is not winning too frequently.  PoET CFT simulates the SGX environment and provides CFT. That said, PoET SGX is preferred because of the additional SGX protections for generating the wait time.

File: generator/source/faq/consensus.rst
Comment (danintel) [reply generator/source/faq/consensus.rst]: Accepted. Copy/paste error.

File: generator/source/faq/consensus.rst
Comment (danintel) [reply generator/source/faq/consensus.rst]: Accepted.

File: generator/source/faq/consensus.rst
Comment (danintel) [reply generator/source/faq/consensus.rst:223]: Accepted.

File: generator/source/faq/consensus.rst
Comment (danintel) [reply generator/source/faq/consensus.rst]: Accepted. Filed https://jira.hyperledger.org/projects/STL/issues/STL-1465 and added a link.

File: generator/source/faq/installation.rst
Comment (danintel) [reply generator/source/faq/installation.rst]: Accepted. It should be `config-genesis.batch`

File: generator/source/faq/installation.rst
Comment (danintel) [reply generator/source/faq/installation.rst]: Accepted. Removed.

File: generator/source/faq/installation.rst
Comment (danintel) [reply generator/source/faq/installation.rst]: Filed https://jira.hyperledger.org/browse/STL-1482
_sawtooth-supply-chain does not support latest release, Sawtooth 1.1 Bionic_

File: generator/source/faq/installation.rst
Comment (danintel) [reply generator/source/faq/installation.rst:265]: Accepted. Filed JIRA issue https://jira.hyperledger.org/projects/STL/issues/STL-1465
and added link to the bug.

File: generator/source/faq/rest.rst
Comment (danintel) [reply generator/source/faq/rest.rst]: Accepted. Reworded answer in _italics_:

> No. There isn't a way to disable that currently because it's determined based on a multiplier of the publishing rate of the network. _Back pressure test is about rejecting load (in the form of batches) that the network wouldn't be able to accommodate in a reasonable amount of time (next couple blocks). This should never be turned off as without this feature it is trivial to do a client-based attack which overwhelms the network creating a DoS_. You would have to make a custom build of Sawtooth to remove that check.

File: generator/source/faq/settings.rst
Comment (danintel) [reply generator/source/faq/settings.rst]: Accepted. Changed 1.2 to _(unreleased) nightly build_.

File: generator/source/faq/validator.rst
Comment (danintel) [reply generator/source/faq/validator.rst]: Accepted. Changed to _the (unreleased) Sawtooth nightly build_.

**Comment by danintel** [comment]
I committed , rebased, and pushed the changes above.

File: generator/source/faq/consensus.rst
Comment (vaporos) [reply generator/source/faq/consensus.rst:22]: Ok. I see how that is relevant for other questions within the FAQ.

**Review by vaporos — APPROVED** [review]

File: generator/source/faq/settings.rst
Comment (agunde406) [inline generator/source/faq/settings.rst:64]: These settings are out of date. please see https://github.com/hyperledger/sawtooth-pbft/pull/85/files#diff-6d8a5e2104954a026de2bfe2b936aea4L78 for the updated ( and soon to be merged) most updated version.

**Review by agunde406 — CHANGES_REQUESTED** [review]

File: generator/source/faq/settings.rst
Comment (danintel) [reply generator/source/faq/settings.rst:64]: I updated the PBFT settings. Thanks!

**Review by agunde406 — APPROVED** [review]
Approving this PR but in the future, please break up the commit into smaller pieces. For example each bullet could be a different commit. That way we dont end up with a ton of commits called "Update FAQ". This would also allow for smaller PRs that are easier to review.

**Comment by danintel** [comment]
Thanks for the review.
The original PR 26 days ago had 3 changes in it.
It grew during the wait for approval.

**Pull request merged**
